src/Microsoft.AspNetCore.Mvc.Core/Internal/AttributeRoute.cs

```diff
@@ -88,48 +88,44 @@ internal void AddEntries(TreeRouteBuilder builder, ActionDescriptorCollection ac
             // action by expected route values, and then use the TemplateBinder to generate the link.
             foreach (var routeInfo in routeInfos)
             {
+                var defaults = new RouteValueDictionary();
+                foreach (var kvp in routeInfo.ActionDescriptor.RouteValues)
+                {
+                    defaults.Add(kvp.Key, kvp.Value);
+                }
+
                 builder.MapOutbound(
-                    _handler,
+                    NullRouter.Instance,
```

Comment:
I'd prefer not to add stuff to routing as part of this change. I'l add a comment.